private/bufpkg/bufimage/util.go

```diff
@@ -71,23 +140,42 @@ func imageWithOnlyPaths(image Image, fileOrDirPaths []string, allowNotExist bool
 		}
 	}
 	if len(potentialDirPaths) == 0 {
-		// we had no potential directory paths as we were able to get
-		// an ImageFile for all fileOrDirPaths, so we can return an Image now
-		// this means we do not have to do the expensive O(image.Files()) operation
-		// to check to see if each file is within a potential directory path
+		// We had no potential directory paths as we were able to get
+		// an ImageFile for all fileOrDirPaths, so we can return an Image now.
+		// This means we do not have to do the expensive O(image.Files()) operation
+		// to check to see if each file is within a potential directory path.
+		//
+		// We do not need to check the excluded paths for the allowNotExist flag because all target
+		// paths were image files, therefore the exclude paths would not apply in this case.
+		// TODO: should we be checking exclude paths that would not apply in this case against
+		// the allowNotExist flag? If so, we would need to do an expensive operation across all
+		// image.Files(). We could have early breaks and returns, similar to the check above, but
+		// this may not be worth it for this case.
```

Comment:
Gah yea this one is a bummer - I think we should still validate that all of the `exclude` paths are valid if `allowNotExist` is not specified. Otherwise users will have silent failures with respect to their `--exclude` flags.

Reply:
Yeah, I've flip-flopped a few times on this one. This is... annoying because we take a nice shortcut for the cases where we only have specific target files because we can just use `if imageFile := image.GetFile(fileOrDirPath); imageFile != nil` or in the case of the module, we just get the storage object directly with `objectInfo, err := sourceReadBucket.Stat(ctx, targetPath)`.

I'm trying to avoid doing a full walk/iterate through all image files just for this verification... but there probably is no way to avoid that...

Reply:
Yea, I see the problem.. it sucks, but we really need to validate this. Otherwise commands like the following will work (I've just tested this on this branch):

```sh
$ buf generate image.bin --path buf/alpha/registry/v1alpha1/authn.proto --exclude_path bogus --template data/template/buf.go.gen.yaml
```

The `--exclude-path` value of `bogus` obviously doesn't exist, but the command continues to work.

We could carve out a separate function that we can call here (and use elsewhere) that simply validates that all of the `exclude-path` flags are valid with respect to the Image. We already do this in the `if len(fileOrDirPaths) == 0 && len(excludeFileOrDirPaths) > 0 {` case above.